src/index.ts

```diff
@@ -3,4 +3,14 @@ import http from 'http';
 
 const server: Server = new Server();
 
-const app: http.Server = server.start();
+let app: http.Server;
+
+server
+  .start()
+  .then((application: http.Server) => {
```

Comment:
How come we aren't using `async/await` here? What's the benefit of using promises over something like 

```js
try {
  const app = await server.start()
} catch (e) {
  console.error(e)
}
```

Reply:
You cant use `await` unless it's in an `async` scope. Believe me I tried. I also tried using an `async` function that used a `try/catch` with `async/await` inside of it, and then calling that function on the app variable you see there. That also doesn't work because an `async` function must return a promise, which would then have to be handled in an `async` function scope, which would return a promise, etc... aka "infinite-loop-never-able-to-escape-async-or-get-a-server-object-returned" `async/await` isn't designed to be used in script style javascript, which is essentially what the entry point of the app is. `.then` works perfectly in this one use case

Reply:
Ah, gotcha. Forgot that we can't use root-level `async/await`. Alright, makes sense to me.